app/views/assignments/_boot.js.erb

```diff
@@ -36,7 +36,7 @@
   function check_required_fields() {
     var invalid = false;
     $('input[required]:not([disabled])').each(function() {
-      if (this.value === '') {
+      if (this.value === '' && this.id !== 'upload_files_for_config') {
```

Comment:
Can you explain the reason for these changes? Maybe I'm missing something?

Reply:
The function here disables all submit functionality unless all required fields are filled. Since the upload modal is on the same page as the normal form and because the zip file is a required field, one cannot submit nor upload an assignment without both filling out the required form fields and providing a zip file to submit. 
This was a quick and dirty way to ignore the upload zip file as a required field for the form as a whole while still making it a required field for the upload modal.